asynction/server.py

```diff
@@ -222,7 +228,10 @@ def _register_handlers(
 
             if channel.x_handlers is not None:
```

Comment:
A big question to ask is what to do if a security scheme is required, but no connect handler is provided. Do we require one be provided when security is required or do we work it out so the security check is run and skips calling the handler?